frontend/util.lua

```diff
@@ -1311,6 +1311,41 @@ function util.stringSearch(txt, str, case_sensitive, start_pos)
     return char_pos
 end
 
+--[[--
+Natural sorting functions, for use with table.sort
+<http://notebook.kulchenko.com/algorithms/alphanumeric-natural-sorting-for-humans-in-lua>
+--]]
+-- Original implementation by Paul Kulchenko
+--[[
+local function addLeadingZeroes(d)
+    local dec, n = string.match(d, "(%.?)0*(.+)")
+    return #dec > 0 and ("%.12f"):format(d) or ("%s%03d%s"):format(dec, #n, n)
+end
+function util.natsort(a, b)
+    return tostring(a):gsub("%.?%d+", addLeadingZeroes)..("%3d"):format(#b)
+           < tostring(b):gsub("%.?%d+", addLeadingZeroes)..("%3d"):format(#a)
+end
+--]]
+-- Hardened (but more expensive) implementation by Egor Skriptunoff, with an UTF-8 tweak by Paul Kulchenko
+local function natsort_conv(s)
+    local res, dot = "", ""
+    for n, m, c in tostring(s):gmatch("(0*(%d*))(.?)") do
+        if n == "" then
+            dot, c = "", dot..c
+        else
+            res = res..(dot == "" and ("%03d%s"):format(#m, m)
+                                   or "."..n)
+            dot, c = c:match("(%.?)(.*)")
+        end
+        res = res..c:gsub("[%z\1-\127\192-\255]", "\0%0")
+    end
+    return res
+end
+function util.natsort(a, b)
+    local ca, cb = natsort_conv(a), natsort_conv(b)
+    return ca < cb or ca == cb and a < b
+end
+
```

Comment:
If you want to speed up things, you can cache a bit :)

My tests showed an improvement (on a Laptop, with 100.000 strings) of `table.sort` from 10.4 seconds to 0.46 seconds, which is not bad :-)

A cached implementation could look like
```
local util.cache = {}
local function natsort(a, b)
    local ca, cb

    if util.cache[a] then
        ca = util.cache[a]
    else
        ca = natsort_conv(a)
        util.cache[a] = ca
    end
    if util.cache[b] then
        cb = util.cache[b]
    else
        cb = natsort_conv(b)
        util.cache[b] = cb
    end

    return ca < cb or ca == cb and a < b
end
```
and to clean up the cache after a sort (just to save memory, can be omitted if it is clear, that only a few entries where sorted)
```
util.cache = {}
```

If you want that automated you can use something like
```
tablesort_orig = table.sort
table.sort = function(array, sort_func)
    tablesort_orig(array, sort_func)
    cache = {}
end
```

Reply:
Yup, that works... *really* well ;).

I tweaked the idea and switched to an LRU cache, which means we don't need to care about maintenance, and the persistence helps the FM use-case.